From: Sitaram, Soundaryaa <Soundaryaa_Sitaram@comcast.com>
Date: 05 Feb 2025 17:24:43 +0000
Subject: [PATCH] ENTOS IDs offset to be adjusted in rdkservices-apis

Upstream-Status: None
Signed-off-by: ssitar583 <Soundaryaa_Sitaram@comcast.com>
---
diff --git a/Source/com/Ids.h b/Source/com/Ids.h
index cf2060333..75d235bbd 100644
--- a/Source/com/Ids.h
+++ b/Source/com/Ids.h
@@ -74,7 +74,8 @@ namespace RPC {
         ID_COMREQUEST_NOTIFICATION      = (ID_OFFSET_INTERNAL + 0x003D),
         ID_SYSTEM_METADATA              = (ID_OFFSET_INTERNAL + 0x003E),

-        ID_EXTERNAL_INTERFACE_OFFSET    = (ID_OFFSET_INTERNAL + 0x0040)
+        ID_EXTERNAL_INTERFACE_OFFSET    = (ID_OFFSET_INTERNAL + 0x0040),
+       ID_EXTERNAL_CC_INTERFACE_OFFSET = (0xCC00) // ends on 0xDFFF
     };
 }
 }
